src/Nethermind/Nethermind.Trie/Pruning/ITrieStore.cs

```diff
@@ -52,6 +52,12 @@ public interface IPruningTrieStore : ITrieStore
         // Acquire lock, then persist and flush cache.
         // Used for full pruning operation that change underlying node storage.
         TrieStore.StableLockScope PrepareStableState(CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Gets the block number of the last persisted block.
+        /// Used to track the oldest available state after full pruning operations.
+        /// </summary>
+        long LastPersistedBlockNumber { get; }
```

Comment:
Are you sure this is the OLDEST persisted block number or just LAST (newest) persisted block number?
Not entirely sure if it solves what we expect it does.
@asdacap can you advise?